lib/grpnames.gi

```diff
@@ -20,46 +20,575 @@
 ##  from GAP3 code written by Markus Püschel and Sebastian Egner.
 ##
 
+
+#############################################################################
+##
+#M  IsTrivialNormalIntersectionInList( <L>, <U>, <V> ) . . . . generic method
+##
+InstallMethod( IsTrivialNormalIntersectionInList,
+               "if minimal normal subgroups are computed",
+               [ IsList, IsGroup, IsGroup ],
+
+  function( L, U, V )
+    local gs;
+
+    gs:= List(Filtered(L, N -> not IsTrivial(N)),
+                                               N -> GeneratorsOfGroup(N)[1]);
+    return ForAll(gs, g -> not g in U or not g in V);
+  end);
+
+#############################################################################
+##
+#M  IsTrivialNormalIntersection( <G>, <U>, <V> ) . . . . . . . generic method
+##
+
+InstallMethod( IsTrivialNormalIntersection,
+               "if minimal normal subgroups are computed",
+               [ IsGroup and HasMinimalNormalSubgroups, IsGroup, IsGroup ],
+
+  function( G, U, V )
+    local gs;
+
+    gs := List(MinimalNormalSubgroups(G), N -> GeneratorsOfGroup(N)[1]);
+    return ForAll(gs, g -> not g in U or not g in V);
```

Comment:
Ah, ok. I actually check in one of these cases if the group itself is trivial (only could come up one place, actually), but I did not think that it is possible to have a nontrivial generator. I will check and correct it. Same question as before applies: should I simply commit and push?